src/test/java/com/puppycrawl/tools/checkstyle/XMLLoggerTest.java

```diff
@@ -63,11 +63,11 @@ public void testEncode()
             {"'", "&apos;"},
             {"\"", "&quot;"},
             {"&", "&amp;"},
-            {"&lt;", "&lt;"},
+            {"&lt;", "&amp;lt;"},
```

Comment:
such change is not expected....
if content is already encoded/escaped ("&lt;") we should skip escaping.

the same for other changes ....

Reply:
@romani You are mistaken the content is not escaped.
The text on the left just that, pure text, which just happens to look like already escaped HTML.
If I wrote Java code like: `System.out.println("5 &lt; 7")` (as an example) we should still print it to the reports as `5 &lt; 7` but escape it since this is XML as `5 &amp;lt; 7`.
If we skip over escaping it, the reports will print `5 < 7` as the text which is not what the original code was.

We deal only with pure text as our violations. Everything in it should be escaped no matter what.

Reply:
good point